Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -149,22 +149,6 @@ protected static string Sanitize(string msg)
       return msg;
     }
 
-    protected void SetupProcess()
-    {
-      Process?.Close();
-      Process = libOptions.CreateSolver(libOptions, options);
-      
-      Process.ErrorHandler += HandleProverError;
-    }
-
-    public override void Close()
```

Comment:
I don't understand the purpose of removing Close from here. It looks like an identical implementation has been moved to the derived classes.

Reply:
Yep, that's intentional. The idea is that the two existing derived classes do it the same way but future derived classes wouldn't need to.

Reply:
Ok, but you could have achieved the same effect by leaving the Close method as is. This method is overridable and derived could change it if necessary. 

In any case, the issue is minor. Think a bit and do what you think is best. I will approve the PR.

Reply:
Yeah, you're right about that. My preference is to require each subclass to implement it separately so that the superclass doesn't seem to prescribe a specific way to start and stop processes, even though it means slightly more code and some duplication. But it's definitely a matter of taste rather than correctness.

If you're happy with that, would you be up for merging it, @shazqadeer? I'm not yet in the right group to be able to do it myself.